### PR TITLE
Add jsdoc blocks to methods

### LINKED ISSUE #10: Add JSDoc
Add documentation to already existing classes in the project so that methods are accounted for when running jsdoc

### PR DESCRIPTION
**Pull request by crokita**
Fixes #10 

This PR is **ready** for review.

### Summary
Add missing documentation to most methods in existing JS files.

### CODE REVIEW

**Comment by crokita** [comment]
Note for future review: This and #12 were part of the same set of changes, but split up. So, some extra documentation will be in #12 instead of this one to prevent merge conflicts between the two branches.

File: lib/js/protocol/BinaryFrameHeader.js [hunk 1]
```diff
@@ -34,6 +34,13 @@ import { BitConverter } from './util/BitConverter.js';
 
 class BinaryFrameHeader {
 
+    /**
+    * @constructor
+    * @param {RpcType} rpcType - A number representing an Enum byte
+    * @param {FunctionID} functionId - A number representing an Enum int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @param {FunctionID} functionId - A FunctionID Enum representing the RPC function that is being described
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 2]
```diff
@@ -34,6 +34,13 @@ import { BitConverter } from './util/BitConverter.js';
...
+    * @param {Number} correlationId - A number representing an int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @param {Number} correlationId - A number representing an ID to correlate a request and response pair
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 3]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
 
-
         return this;
 
     }
 
+    /**
+    * Takes in a byte array and transforms it to an instance of this class
+    * @param {Uint8Array} binHeader
+    * @return {BinaryFrameHeader}
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @return {BinaryFrameHeader} an instance of this class from parsing the supplied Uint8Array 
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 4]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
 
+    /**
+    * @param {Uint8Array} type - A number representing an Enum byte
+    */
     setRPCType(type) {
         this._rpcType = type;
     }
+    
+    /**
+    * @return {RpcType} - A number representing an Enum byte
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @return {RpcType} rpcType - An RPC Enum  representing the type of RPC being described (Request, Response, Notification)
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 5]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
+    */
     getRPCType() {
         return this._rpcType;
     }
-
+    
+    /**
+    * @param {RpcType} id - A number representing an Enum int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @param {RpcType} rpcType - An RPC Enum  representing the type of RPC being described (Request, Response, Notification)
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 6]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
+    /**
+    * @return {FunctionID} - A number representing an Enum int
+    */
     getFunctionId() {
         return this._functionId;
     }
-
+    
+    /**
+    * @param {FunctionID} id - A number representing an int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @param {FunctionID} functionId - A FunctionID Enum representing the RPC function that is being described
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 7]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
+    
+    /**
+    * @param {RpcType} id - A number representing an Enum int
+    */
     setFunctionId(id) {
         this._functionId = id;
     }
+    
+    /**
+    * @return {FunctionID} - A number representing an Enum int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @return {FunctionID} functionId - A FunctionID Enum representing the RPC function that is being described
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 8]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
+    /**
+    * Uses the private members of this class to construct a byte array header
+    * @return {Uint8Array} - A byte array
+    */
     assembleHeaderBytes() {
         // TODO
     }
 
+    /**
+    * @param {Uint8Array} type - A number representing an Enum byte
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @param {RpcType} rpcType - An RPC Enum  representing the type of RPC being described (Request, Response, Notification)
```

File: lib/js/protocol/BinaryFrameHeader.js [hunk 9]
```diff
@@ -44,57 +51,107 @@ class BinaryFrameHeader {
...
+    
+    /**
+    * @param {FunctionID} id - A number representing an int
+    */
     setCorrelationId(id) {
         this._correlationId = id;
     }
+    
+    /**
+    * @return {Number} - A number representing an int
```
Comment (joeygrover) [inline lib/js/protocol/BinaryFrameHeader.js]:
```suggestion
    * @return {Number} correlationId - A number representing an ID to correlate a request and response pair

```

File: lib/js/protocol/SdlPacket.js [hunk 10]
```diff
@@ -65,7 +99,18 @@ class SdlPacket {
     //RESERVED 0x10 - 0xFF
 
     /**
-    * Frame Info
+    * @constructor
+    * @param {Number} version - Protocol version to use
+    * @param {Boolean} encryption - Whether or not this packet is encrypted
```
Comment (joeygrover) [inline lib/js/protocol/SdlPacket.js]:
```suggestion
    * @param {Boolean} encryption - Whether or not the payload in this packet is encrypted
```

File: lib/js/protocol/SdlPacket.js [hunk 11]
```diff
@@ -65,7 +99,18 @@ class SdlPacket {
...
+    * @param {FrameType} frameType - A number representing the packet frame type
+    * @param {ServiceType} serviceType - The service that this packet is associated with
+    * @param {Number} frameInfo - Specific frame info related to this packet
+    * @param {Number} sessionID - ID this packet is associated with
+    * @param {Number} dataSize - Size of the payload that will be added
+    * @param {Number} messageID - ID of this specific packet
+    * @param {Uint8Array} payload - Raw data that will be attached to the packet (RPC message, raw bytes, etc)
+    * @param {Number} offset
+    * @param {Number} bytesToWrite
     */
```
Comment (joeygrover) [inline lib/js/protocol/SdlPacket.js]: Is this the right location for this?

File: lib/js/protocol/SdlPacket.js [hunk 12]
```diff
@@ -108,45 +153,79 @@ class SdlPacket {
 
     }
 
+    /**
+    * @return {Number} - Protocol version to use
```
Comment (joeygrover) [inline lib/js/protocol/SdlPacket.js]:
```suggestion
    * @return {Number} - Protocol version used by this packet
```

File: lib/js/protocol/SdlPacket.js [hunk 13]
```diff
@@ -108,45 +153,79 @@ class SdlPacket {
...
+    */
     getVersion() {
         return this._version;
     }
 
+    /**
+    * @return {Boolean} - Whether or not this packet is encrypted
```
Comment (joeygrover) [inline lib/js/protocol/SdlPacket.js]:
```suggestion
    * @return {Boolean} - Whether or not the payload in this packet is encrypted
```

File: lib/js/protocol/SdlPacket.js [hunk 14]
```diff
@@ -180,16 +266,16 @@ class SdlPacket {
...
-	 * @param frameType the packet frame type
-	 * @param serviceType the service that this packet is associated with
-	 * @param controlFrameInfo specific frame info related to this packet
-	 * @param sessionID ID this packet is associated with
-	 * @param dataSize size of the payload that will be added
-	 * @param messageID ID of this specific packet
-	 * @param payload raw data that will be attached to the packet (RPC message, raw bytes, etc)
-	 * @return a byte[] representation of an SdlPacket built using the supplied params
+     * @param {Number} version - Protocol version to use
+     * @param {Boolean} encryption - Whether or not this packet is encrypted
```
Comment (joeygrover) [inline lib/js/protocol/SdlPacket.js]:
```suggestion
     * @param {Boolean} encryption - Whether or not the payload in this packet is encrypted
```

**Pull request merged**